### PR TITLE
Fix conversion for text <-> list, and quote <-> list

### PR DESCRIPTION
**Pull request by notnownikki**
Converts text and quotes to list by using BR as the delimiter for list items. Transformations have been moved into the list block, it seemed to make sense for them to live there as the transformation code is there too.

Fixes https://github.com/WordPress/gutenberg/issues/1214

### CODE REVIEW

File: blocks/library/list/index.js [hunk 1]
```diff
@@ -15,6 +15,42 @@ import BlockControls from '../../block-controls';
 
 const { children, prop } = hpq;
 
+const fromBrDelimitedContent = function( content ) {
+	const listItems = [];
+	listItems.push( createElement( 'li', [], [] ) );
+	content.forEach( function( element, elementIndex, elements ) {
```
Comment (BE-Webdesign) [inline blocks/library/list/index.js]: Minor Nitpick: I think we might want to use arrow function for these callbacks to keep consistent.

File: blocks/library/list/index.js [hunk 2]
```diff
@@ -15,6 +15,42 @@ import BlockControls from '../../block-controls';
...
+			listItems[ listItems.length - 1 ].props.children.push( element );
+		}
+	} );
+	return listItems;
+};
+
+const toBrDelimitedContent = function( values ) {
+	const content = [];
+	values.forEach( function( li, liIndex, listItems ) {
+		if ( Array.isArray( li.props.children ) ) {
```
Comment (BE-Webdesign) [inline blocks/library/list/index.js]: Another minor idea: I think we could potentially condense this code by using the [`Children` API](https://facebook.github.io/react/docs/react-api.html#react.children) in React. Either by making use of `Children.map` or `Children.toArray`, then the conditional checking for the array won't really be necessary.

**Review by BE-Webdesign — APPROVED** [review]
Looks Great! Works really well, left some minor feedback, which I would love to hear your thoughts on!

**Comment by BE-Webdesign** [comment]
Hmm, actually while circling back to this, I did come across a weird situation. If things get indented, then when you transform the indented children never get changed to a text or quote properly.

**Comment by notnownikki** [comment]
Ah, when a user likes lists so much that they put a list in their list so they can list their lists... Yeah, I'm not sure that we can preserve the indentation when we convert to a text or quote block, but we should certainly get the values of the list items.

**Comment by notnownikki** [comment]
I've added support for nested lists now. Also, there were bugs when converting from empty blocks, so they're fixed too.

**Review by BE-Webdesign — APPROVED** [review]
:shipit:

**Pull request merged**